Makefile

```diff
@@ -93,129 +92,49 @@ release: certs docker-release
-else
-GO_INTEG_TEST = go test -race -tags integration -c -o
+# -race sometimes causes compile issues on Arm
+ifneq (${BUILD_PLATFORM},aarch64)
+	GOTEST += -race
 endif
 
-out/test-artifacts/linux-engine-tests: $(ALL_GO_FILES) .out-stamp .builder-image-stamp
-	$(call dockerbuild,$(GO_INTEG_TEST) $@ ./agent/engine)
-
-out/test-artifacts/linux-app-tests: $(ALL_GO_FILES) .out-stamp .builder-image-stamp
-	$(call dockerbuild,$(GO_INTEG_TEST) $@ ./agent/app)
-
-out/test-artifacts/linux-stats-tests: $(ALL_GO_FILES) .out-stamp .builder-image-stamp
-	$(call dockerbuild,$(GO_INTEG_TEST) $@ ./agent/stats)
-
-out/test-artifacts/windows-engine-tests.exe: $(ALL_GO_FILES) .out-stamp .builder-image-stamp
-	$(call win-cgo-dockerbuild,$(GO_INTEG_TEST) $@ ./agent/engine)
-
-out/test-artifacts/windows-app-tests.exe: $(ALL_GO_FILES) .out-stamp .builder-image-stamp
-	$(call win-cgo-dockerbuild,$(GO_INTEG_TEST) $@ ./agent/app)
-
-out/test-artifacts/windows-stats-tests.exe: $(ALL_GO_FILES) .out-stamp .builder-image-stamp
-	$(call win-cgo-dockerbuild,$(GO_INTEG_TEST) $@ ./agent/stats)
-
-GO_FUNCTIONAL_TEST = go test -tags functional -c -o
-out/test-artifacts/linux-simple-tests: $(ALL_GO_FILES) .out-stamp .builder-image-stamp
-	$(call dockerbuild,$(GO_FUNCTIONAL_TEST) $@ ./agent/functional_tests/tests/generated/simpletests_unix)
-
-out/test-artifacts/linux-handwritten-tests: $(ALL_GO_FILES) .out-stamp .builder-image-stamp
-	$(call dockerbuild,$(GO_FUNCTIONAL_TEST) $@ ./agent/functional_tests/tests)
+test:
+	${GOTEST} -tags unit -timeout=30s ./agent/...
 
-out/test-artifacts/windows-simple-tests.exe: $(ALL_GO_FILES) .out-stamp .builder-image-stamp
-	$(call win-cgo-dockerbuild,$(GO_FUNCTIONAL_TEST) $@ ./agent/functional_tests/tests/generated/simpletests_windows)
+test-silent:
+	$(eval undefine VERBOSE)
+	${GOTEST} -tags unit -timeout=30s ./agent/...
 
-out/test-artifacts/windows-handwritten-tests.exe: $(ALL_GO_FILES) .out-stamp .builder-image-stamp
-	$(call win-cgo-dockerbuild,$(GO_FUNCTIONAL_TEST) $@ ./agent/functional_tests/tests)
-
-##.PHONY: test-artifacts-windows test-artifacts-linux test-artifacts
-
-WINDOWS_ARTIFACTS_TARGETS := out/test-artifacts/windows-engine-tests.exe out/test-artifacts/windows-stats-tests.exe
-WINDOWS_ARTIFACTS_TARGETS += out/test-artifacts/windows-app-tests.exe out/test-artifacts/windows-simple-tests.exe
-WINDOWS_ARTIFACTS_TARGETS += out/test-artifacts/windows-handwritten-tests.exe
-
-LINUX_ARTIFACTS_TARGETS := out/test-artifacts/linux-engine-tests out/test-artifacts/linux-stats-tests
-LINUX_ARTIFACTS_TARGETS += out/test-artifacts/linux-app-tests out/test-artifacts/linux-simple-tests
-LINUX_ARTIFACTS_TARGETS += out/test-artifacts/linux-handwritten-tests
-
-test-artifacts-windows: $(WINDOWS_ARTIFACTS_TARGETS)
-
-test-artifacts-linux: $(LINUX_ARTIFACTS_TARGETS)
+run-integ-tests: test-registry gremlin container-health-check-image run-sudo-tests
+	ECS_LOGLEVEL=debug ${GOTEST} -tags integration -timeout=25m ./agent/...
```

Comment:
Are we skipping the `-race` flag? 🤔

Reply:
Nope. It gets added conditionally to $GOTEST